src/Events/Extensions/UriExtensions.cs

```diff
@@ -12,6 +12,8 @@ namespace Altinn.Platform.Events.Extensions
     /// </summary>
     public static class UriExtensions
     {
+        private static string _urnPattern = @"\burn:[a-z0-9][a-z0-9-]{0,31}:[a-z0-9()+,\-.:=@;$_!*'%/?#]+";
```

Comment:
This will allow strings like `urn:1:::::""""` which we probably don't want.

```suggestion
        private static string _urnPattern = @"^urn:[a-z][a-z0-9-]{0,31}(:[a-z][a-z0-9()+,.\-=@;$_!*'%#]{0,99}){1,10}\/?$";
```
* Add `$` to disallow arbitrary trailing characters
* Replace `\b` with `^` to disallow whitespace at the beginning
* Require all "parts" to begin with a letter (ie. disallow "urn:1:2")
* Limit each "part" beyond the first to max 100 characters
* Disallow empty "parts" (ie. `urn:altinn:foo::::`)
* Require 1 to 10 parts beyond the first

Reply:
Change seems to be breaking some of the unit tests. Will have a look

Reply:
`urn:isbn:0451450523` is a valid urn ref https://en.wikipedia.org/wiki/Uniform_Resource_Name.
Rolling back to existing regex for now. Consider reworking the regex so existing unit tests run green @elsand ?

Reply:
Could `System.Uri.IsWellFormedUriString`  with UriKind.Absolute and a check for scheme `urn` be an alternative?

https://learn.microsoft.com/en-us/dotnet/api/system.uri.iswellformeduristring?view=net-7.0

Reply:
Did a quick test and `urn:altinn:foo:::::` passes the check for System.Uri.IsWellFormedUriString. I believe we verify that it is a well formed uri before we reach this point as well.

Reply:
That being said.. if Microsoft deems it a well formed uri, we might be happy enough with that? Easier than maintaining the regex string at least.

Reply:
`^urn:[a-z0-9-]{1,30}(:[a-z0-9()+,.\-=@;$_!*'%#\/]{1,100}){1,10}$` will allow numerics as the first (or sole) character in both NID and NSS parts. This also includes `/` as a valid NSS character, which seems reasonable (as we allow it as a trailing character)

We should perhaps consider restricting our support to a smaller subset of that standard ref wikipedia. I'm not sure it will ever make sense here to support percent-encoded arbitrary bytes in the NSS part, or query-components and fragments. I'm not too confident allowing eg. NUL bytes (even encoded) won't have any potential security implications down the line.

`^urn:[a-z0-9-]{1,30}(:[a-z0-9()+,.\-;$_!\/]{1,100}){1,10}$` disallows `% ? #` while still passing the unit tests and support a reasonably wide range of subdivisions in the NSS-es.

On the other hand, if we _are_ to support query-components and fragments, I think we should attempt to validate that those are well-formed (perhaps by explicitly calling `IsWellFormedUriString`)

Reply:
If `System.Uri.IsWellFormedUriString` is _that_ lax, then for sure, being more restrictive sounds like a good idea. Could maybe add some test coverage for empty URN segments, to make sure we don't have unintended wildcard matching in the middle of a URN path.

Reply:
I think we can follow up on this at a later point.. Regardless the URN they choose for the resource will have to pass checks when being registered in the resource registry as well. We will sync with them on the validation.